### ISSUE TITLE
autoBuildFlags does not impact the install

### ISSUE THREAD

**Issue by piercus** [UrielCh/opencv4nodejs#73]
## Context

I'm working on building opencv4nodejs in a lambda layer, the project can be found in https://github.com/piercus/opencv4nodejs-lambda-layer

Lambda layer maximal size is 250MB. I cannot put the whole opencv build in it.
Currently, i want to limit the layer to "core" and "calib3d".

I have set the corresponding flags in
https://github.com/piercus/opencv4nodejs-lambda-layer/blob/master/guest/package.json#L13

```
    "autoBuildFlags": "-DCMAKE_BUILD_TYPE=RELEASE -DBUILD_LIST=core,calib3d -DWITH_IPP=OFF -DWITH_V4L=OFF -DWITH_FFMPEG=OFF -DWITH_GSTREAMER=OFF -DWITH_GTK=OFF -DWITH_PROTOBUF=OFF -DBUILD_PROTOBUF=OFF -DOPENCV_DNN_OPENCL=OFF -DWITH_WIN32UI=OFF -DWITH_PNG=OFF -DWITH_JPEG=OFF -DWITH_TIFF=OFF -DWITH_WEBP=OFF -DWITH_OPENJPEG=OFF -DWITH_JASPER=OFF -DWITH_OPENEXR=OFF -DBUILD_opencv_python=OFF -DBUILD_opencv_java=OFF"
```

The packaging is made in a [github action](https://github.com/piercus/opencv4nodejs-lambda-layer/blob/master/.github/workflows/release.yml) : see the [job](https://github.com/piercus/opencv4nodejs-lambda-layer/actions/runs/4151373985/jobs/7181646697)

Inside a docker which is doing [npm install](https://github.com/piercus/opencv4nodejs-lambda-layer/blob/master/guest/package-guest-install.bash)

## Actual

The resulting [asset](https://github.com/piercus/opencv4nodejs-lambda-layer/releases/tag/cv-v4.6.0) is 312MB big and contains all the libs 

## Expected

The resulting asset should contain only calib3d and core libs

**Comment by UrielCh** [comment]
Oups... 
I will check that soon.

**Comment by UrielCh** [comment]
I find the code that is incorrect, but that will not change the result.

to build the openCV the way you want the current implementation, you should replace :

`-DBUILD_LIST=core,calib3d` with:

```bash
-DBUILD_opencv_core=ON
-DBUILD_opencv_calib3d=ON
-DBUILD_opencv_python_tests=OFF
-DBUILD_opencv_core=OFF
-DBUILD_opencv_imgproc=OFF
-DBUILD_opencv_imgcodecs=OFF
-DBUILD_opencv_videoio=OFF
-DBUILD_opencv_highgui=OFF
-DBUILD_opencv_video=OFF
-DBUILD_opencv_calib3d=OFF
-DBUILD_opencv_features2d=OFF
-DBUILD_opencv_objdetect=OFF
-DBUILD_opencv_dnn=OFF
-DBUILD_opencv_ml=OFF
-DBUILD_opencv_flann=OFF
-DBUILD_opencv_photo=OFF
-DBUILD_opencv_gapi=OFF
-DBUILD_opencv_face=OFF
-DBUILD_opencv_img_hash=OFF
-DBUILD_opencv_text=OFF
-DBUILD_opencv_tracking=OFF
-DBUILD_opencv_videostab=OFF 
-DBUILD_opencv_xfeatures2d=OFF
-DBUILD_opencv_ximgproc=OFF
```

Should -DBUILD_LIST have a higher priority than the -DBUILD_opencv ?

and should a -DBUILD_LIST disable all build -DBUILD flags when present?

I do not have access to my regular PC, I only have access to a M1 Mac. now,

**Comment by piercus** [comment]
Thanks for the quick response @UrielCh !

I will confirm in few hours that this is working!

**Comment by UrielCh** [comment]
I made a patch to ensure that this approach works, but I did not commit it yet

**Comment by UrielCh** [comment]
I'm waiting for your next feed back then I will publish a first patched version.

**Comment by piercus** [comment]
I confirm that changes suggested in https://github.com/UrielCh/opencv4nodejs/issues/73#issuecomment-1427076308 are not changing the end result.

See https://github.com/piercus/opencv4nodejs-lambda-layer/releases/tag/layer-cv-v4.6.0

If the patch is available in a specific branch, tell me which one so i will test.

**Comment by piercus** [comment]
Still not able to make it work (see [the job](https://github.com/piercus/opencv4nodejs-lambda-layer/actions/runs/4167222585/jobs/7212533145) and [the release](https://github.com/piercus/opencv4nodejs-lambda-layer/releases/tag/layer-4.6.0))

Even when using `--verbose`, I'm still not able to get the logs of the `npm install` command.
I have made few attemps, but if you have a tips to get the logs displayed it would help me give you more infos.

**Comment by UrielCh** [comment]
My internet connection quality is too low to test my changes.
but using your first `autoBuildFlags` should output those new lines:

```bash
info config cmake flag contains "-DBUILD_LIST=core,calib3d" automatic cmake flags are now disabled.
info config adding cmake flag "-DCMAKE_BUILD_TYPE=RELEASE"
info config adding cmake flag "-DBUILD_LIST=core,calib3d"
info config adding cmake flag "-DWITH_IPP=OFF"
info config adding cmake flag "-DWITH_V4L=OFF"
info config adding cmake flag "-DWITH_FFMPEG=OFF"
info config adding cmake flag "-DWITH_GSTREAMER=OFF"
info config adding cmake flag "-DWITH_GTK=OFF"
info config adding cmake flag "-DWITH_PROTOBUF=OFF"
info config adding cmake flag "-DBUILD_PROTOBUF=OFF"
info config adding cmake flag "-DOPENCV_DNN_OPENCL=OFF"
info config adding cmake flag "-DWITH_WIN32UI=OFF"
info config adding cmake flag "-DWITH_PNG=OFF"
info config adding cmake flag "-DWITH_JPEG=OFF"
info config adding cmake flag "-DWITH_TIFF=OFF"
info config adding cmake flag "-DWITH_WEBP=OFF"
info config adding cmake flag "-DWITH_OPENJPEG=OFF"
info config adding cmake flag "-DWITH_JASPER=OFF"
info config adding cmake flag "-DWITH_OPENEXR=OFF"
info config adding cmake flag "-DBUILD_opencv_python=OFF"
info config adding cmake flag "-DBUILD_opencv_java=OFF"
```

try to check your `auto-build.json` in the `${OPENCV_BUILD_ROOT}/version/`

the file contains the effective `cmake` config parameters.

**Comment by piercus** [comment]
For the record : 

In order to get the logs i need to do 

```
npm install --foreground-scripts
```

see https://github.com/npm/cli/issues/3347

**Comment by piercus** [comment]
fixed in v6.4.3 thanks @UrielCh
